pkg/resource/providerconfig.go

```diff
@@ -95,6 +95,8 @@ func CommonCredentialExtractor(ctx context.Context, source xpv1.CredentialsSourc
 		return ExtractFs(ctx, afero.NewOsFs(), selector)
 	case xpv1.CredentialsSourceSecret:
 		return ExtractSecret(ctx, client, selector)
+	case xpv1.CredentialsSourceInjectedIdentity:
```

Comment:
Given we don't have to extract any credentials, in this case, we can check whether the source equals `xpv1.CredentialsSourceInjectedIdentity` before calling `CommonCredentialExtractor`. 
That approach makes it more explicit regarding the intent.

See [this](https://github.com/crossplane-contrib/provider-helm/blob/a51a31a6a4f8c9ce31503a0c48513b089cb2b929/pkg/controller/release/release.go#L171) as an example.